src/main/java/org/sonarlint/intellij/ui/LastAnalysisPanel.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * SonarLint for IntelliJ IDEA
+ * Copyright (C) 2015 SonarSource
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonarlint.intellij.ui;
+
+import com.intellij.openapi.Disposable;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.Disposer;
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.awt.Insets;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import javax.swing.Box;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.Timer;
+import org.sonarlint.intellij.issue.ChangedFilesIssues;
+import org.sonarlint.intellij.util.ResourceLoader;
+import org.sonarlint.intellij.util.SonarLintUtils;
+
+public class LastAnalysisPanel implements Disposable {
+  private static final String NO_ANALYSIS_LABEL = "Trigger the analysis to find issues on the files in the change set";
+  private final ChangedFilesIssues changedFileIssues;
+  private GridBagConstraints gc;
+  private Timer lastAnalysisTimeUpdater;
+  private JLabel lastAnalysisLabel;
+  private JLabel icon;
+  private JPanel panel;
+
+  public LastAnalysisPanel(ChangedFilesIssues changedFileIssues, Project project) {
+    this.changedFileIssues = changedFileIssues;
+    createComponents();
+    setLastAnalysisTime();
+    setTimer();
+    Disposer.register(project, this);
+  }
+
+  public JPanel getPanel() {
+    return panel;
+  }
+
+  public void update() {
+    setLastAnalysisTime();
+  }
+
+  private void setLastAnalysisTime() {
+    LocalDateTime lastAnalysis = changedFileIssues.lastAnalysisDate();
+    panel.removeAll();
+    if (lastAnalysis == null) {
+      lastAnalysisLabel.setText(NO_ANALYSIS_LABEL);
+      panel.add(icon);
+      panel.add(lastAnalysisLabel, gc);
+    } else {
+      lastAnalysisLabel.setText("Analysis done " + SonarLintUtils.age(System.currentTimeMillis()));
+      panel.add(lastAnalysisLabel, gc);
+    }
+
+    panel.add(Box.createHorizontalBox(), gc);
+  }
+
+  private void createComponents() {
+    panel = new JPanel(new GridBagLayout());
+    try {
+      icon = new JLabel(ResourceLoader.getIcon("info.png"));
+    } catch (IOException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1166)